### PR TITLE
Library

### PR DESCRIPTION
**Pull request by thsvr**
Hello TSE reviewing this pull request.
We have completed the requirements for the first Javascript project:

- Book constructor
- addBookToLybrary function that adds book instances to the library array
- helper functions to edit book read status and delete book from library
- persistent local library via localstorage
- user interface for adding, editing status and deleting books

Thanks for your time,
Sarah & Thais 🦄

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -9,10 +9,12 @@
       "sourceType": "module"
     },
     "extends": ["airbnb-base"],
+    "ignorePatterns": ["*.json"],
```
Comment (bellom) [inline .eslintrc.json]: You made some changes in this file, check out the [microverse-linter-config](https://github.com/microverseinc/linters-config/tree/master/javascript)
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/

File: index.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
+const booksSection = document.getElementById('books');
+const localLibrary = localStorage.getItem('myLibrary');
+
+// get latested localStorage values
```
Comment (bellom) [inline index.js]: It’s better to delete commented code. If you need to review a previous version of your code, use git history, your code is clear enough to understand 🙂 ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

Kindly remove all commented code

File: index.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+// parse string into array
+const getStoredLibrary = () => JSON.parse(localStorage.getItem('myLibrary'));
+
+const storeLibrary = (library) => {
+  // parse array into string and save it again in the locastorage
+  localStorage.setItem('myLibrary', JSON.stringify(library));
+};
+
+const bookAdder = (book, index) => `<article class='book' id='book-${index}'>
+  <h3>${book.title}</h3>
```
Comment (bellom) [inline index.js:5]: Let separate the DOM from application logic, you can create another file for DOM.
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59623451/

File: index.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  const updateLibrary = getStoredLibrary();
+  // take item out of array
+  updateLibrary.splice(index, 1);
+  storeLibrary(updateLibrary);
+  document.getElementById(`book-${index}`).outerHTML = '';
+};
+
+const updateBook = (index, field) => {
+  const updateLibrary = getStoredLibrary();
+  // update value of read as oposite of current value
```
Comment (bellom) [inline index.js]: It’s better to delete commented code. If you need to review a previous version of your code, use git history, your code is clear enough to understand 🙂 ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

Kindly remove all commented code

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome Design :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Below are part of the checklist for this project, kindly work on these few things listed below;**

**_1. Kindly attend to my comments below_**

**_2. A user shouldn't be able to add a book with empty value right._**
image

**_3. Kindly update your `README` with a live link, `README` file without live link doesn't look professional._**
Example: https://raw.githack.com/uryelah/-Microverse-601-Library/library/index.html
Kindly watch this video tutorial on how to create live link below 👇;
[Video Tutorial](https://www.loom.com/share/76088842a2d14322bbcf65b884f4d42d)
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/

**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

Best,

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: index.js (hunk 2 above)
Comment (uryelah) [reply index.js]: That's a comment describing the function bellow, we'll rewrite it to read better like a normal comment

File: index.js (hunk 4 above)
Comment (uryelah) [reply index.js]: That's a comment describing the function bellow, we'll rewrite it to read better like a normal comment

**Comment by uryelah** [comment]
**Hey there!** 
We did the requested changes from the last review.
The cover link input is optional because we have a default cover set, just in case, in the constructor.
Can you take a look if it's enough for approval? 
We are open to feedback 🙂

**P.S.:**
We, as suggested by Bello, added `/* eslint-disable no-unused-vars, no-unused-vars */` comments to the functions being called inside `<tag onclick='function'>` and on the logic.js file due to eslint not being able to recognize that they were being used in the html and/or at the other script file.

**Review by bolah2009 — APPROVED** [review]
It's a 👍🏾 from me!

### :heavy_check_mark: Approved!

Good luck with your next projects. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Pull request merged**